File: bootstrap-ui-tests/src/test/ruby/pages/messages_page.rb
class MessagesPage < SitePrism::Page
    url = ENV['BOOTSTRAP_URL'] || 'localhost:8080'

	set_url "http://#{ url }/#/"
	
	element :message_field, "textarea"
	element :send_button, "input[type=submit]"
	element :logout_link, "a[text()='Logout']"
	
	def send_message(message)
		message_field.set(message)
		send_button.click
	end
	
	def delete_message(message)
		find(:xpath, "//p[text()='#{message}']/following-sibling::span").click_on("Delete")
	end
	
	def edit_message(message)
		find(:xpath, "//p[text()='#{message}']/following-sibling::span").click_on("Edit")
	end
	
	def logout()
		logout_link.click
	end
	
end

File: ui-tests/src/test/scala/uitest/pages/MessagesPage.scala
package uitest.pages

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy
import uitest.commands.SeleniumCommands

class MessagesPage(driver: WebDriver) {
  private val sc: SeleniumCommands = new SeleniumCommands(driver)

  @FindBy(css = "#logoutLink")
  val logoutLink: WebElement = null

  @FindBy(css = ".alert-info")
  val infoAlert: WebElement = null

  @FindBy(css = ".alert-danger")
  val errorAlert: WebElement = null

  @FindBy(css = "textarea")
  val messageField: WebElement = null

  @FindBy(css = "input[type=submit]")
  val sendButton: WebElement = null

  def logout() {
    logoutLink.click()
    sc.waitForFinishLoading()
  }

  def isUserLogged(user: String): Boolean = {
    sc.waitForElementVisible(By.linkText("Logged in as " + user))
    true
  }

  def isUMessageDisplayed(message: String): Boolean = {
    val escapedMessage = message.replace("\"", "\\\"")
    sc.waitForElementVisible(By.xpath(s"""//span[contains(., "$escapedMessage")]"""))
    true
  }

  def getInfoText: String = {
    sc.waitForElementVisible(By.cssSelector(".alert-info"))
    infoAlert.getText
  }

  def getErrorText: String = {
    sc.waitForElementVisible(By.cssSelector(".alert-danger"))
    errorAlert.getText
  }
}
